processing/src/main/java/io/druid/segment/SegmentMetadataHelper.java

```diff
@@ -0,0 +1,63 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.segment;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import io.druid.jackson.DefaultObjectMapper;
+import io.druid.query.aggregation.AggregatorFactory;
+
+import java.util.Map;
+
+/**
+ */
+public class SegmentMetadataHelper
+{
+  public static final String METRIC_SPEC_KEY = "metricsSpec";
+
+  private static final ObjectMapper mapper = new DefaultObjectMapper();
```

Comment:
It looks to me IndexMaker and IndexMerger both implement same functions, wouldn't it make more sense for both of them to inherit from a base class, and create shared static injector there? We can also put SegmentMetadataHelper as an inner class of the base class since only IndexMaker and IndexMerger use it.